src/main/java/com/example/repository/user/UserRepository.java

```diff
@@ -0,0 +1,17 @@
+package com.example.repository.user;
+
+import com.example.model.User;
+import java.util.Optional;
+import org.springframework.data.jpa.repository.EntityGraph;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+
+public interface UserRepository extends JpaRepository<User, Long> {
+
+    @EntityGraph(attributePaths = "roles")
+    @Query("FROM User u where u.email=:email")
+    Optional<User> findByEmail(@Param("email") String email);
```

Comment:
```suggestion
    Optional<User> findByEmail(String email);
```

Reply:
intelej idea doesn't hint me the name of method. That's why I wrote @Query

Reply:
How to know JpaRepository can generate such method or not?